src/Mappers/BaseMapper.php

```diff
@@ -62,17 +64,158 @@ protected static function getValueFromTag(\DOMDocument $document, string $tag):
     {
         $fieldElement = $element->getElementsByTagName($fieldTagName)->item(0);
+
         if (!isset($fieldElement)) {
             return null;
         }
 
+        if (isset($object)) {
+            self::checkForMessage($object, $fieldElement);
+        }
+
         if ($fieldElement->textContent === "") {
             return null;
         }
 
         return $fieldElement->textContent;
     }
+    
+    protected static function checkForMessage($object, \DOMElement $element): void
+    {
+        if ($element->hasAttribute('msg')) {
+            $message = new Message();
+            $message->setType($element->getAttribute('msgtype'));
+            $message->setMessage($element->getAttribute('msg'));
+            $message->setField($element->nodeName);
+
+            $object->addMessage($message);
+        }
+    }
+
+    protected static function getAttribute(\DOMElement $element, string $fieldTagName, string $attributeName): ?string
+    {
+        $fieldElement = $element->getElementsByTagName($fieldTagName)->item(0);
+
+        if (!isset($fieldElement)) {
+            return null;
+        }
+
+        if ($fieldElement->getAttribute($attributeName) === "") {
+            return null;
+        }
+
+        return $fieldElement->getAttribute($attributeName);
+    }
+
+    protected static function parseBooleanAttribute(?string $value): ?bool
+    {
+        return filter_var($value, FILTER_VALIDATE_BOOLEAN);
+    }
+
+    protected static function parseDateAttribute(?string $value): ?\DateTimeImmutable
+    {
+        if ((bool)strtotime($value)) {
+            return Util::parseDate($value);
+        } else {
+            return null;
+        }
+    }
+
+    protected static function parseDateTimeAttribute(?string $value): ?\DateTimeImmutable
+    {
+        if ((bool)strtotime($value)) {
+            return Util::parseDateTime($value);
+        } else {
+            return null;
+        }
+    }
+
+    protected static function parseEnumAttribute(string $enumName, ?string $value)
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        $enum = "\\PhpTwinfield\\Enums\\" . $enumName;
+
+        try {
+            $classReflex = new \ReflectionClass($enum);
+            $classConstants = $classReflex->getConstants();
+
+            foreach ($classConstants as $classConstant) {
+                if ($value == $classConstant) {
+                    return new $enum($value);
+                }
+            }
+        } catch (\ReflectionException $e) {
+            return null;
+        }
+
+        return null;
+    }
+
+    protected static function parseMoneyAttribute(?float $value): ?Money
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        return Util::parseMoney($value, new Currency('EUR'));
+    }
+
+    protected static function parseObjectAttribute(string $className, $object, \DOMElement $element, string $fieldTagName, array $attributes = null)
```

Comment:
Based on the earlier comments, can you update this method?

Reply:
Done